### PR TITLE
[AIRFLOW-1124] Do not set all tasks to scheduled in backfill

### PR DESCRIPTION
**Pull request by bolkedebruin**
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [X] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1124

### Description
- [X] Here are some details about my PR, including screenshots of any UI changes:

Backfill is supposed to fill in the blanks and not to reschedule
all tasks. This fixes a regression from 1.8.0.

### Tests
- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

- Tests added

### Commits
- [X] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

@mistercrunch @aoen @criccomini

### CODE REVIEW

File: airflow/jobs.py [hunk 1]
```diff
@@ -1821,7 +1821,8 @@ def _execute(self):
 
             for ti in run.get_task_instances():
                 # all tasks part of the backfill are scheduled to run
-                ti.set_state(State.SCHEDULED, session=session)
+                if ti.state == State.NONE:
```
Comment (artwr) [inline airflow/jobs.py:1824]: Given Max's comment I think that's fair but what about failed tasks?

**Review by artwr — APPROVED** [review]
LGTM +1

File: airflow/jobs.py (hunk 1 above)
Comment (bolkedebruin) [reply airflow/jobs.py:1824]: @artwr good question. Should they be retried or should the user clear them first? 1.7.1 seems to re-evaluate them, but is a bit ambiguous in other areas: for example a task instance set to SKIPPED is being skipped again, while the underlying condition might have a different outcome now.

btw: failed tasks are being re-evaluated in my code as well, but will be immediately kicked out as they failed earlier.

Comment (artwr) [reply airflow/jobs.py:1824]: A very good question. For the use cases I have in mind, anything that is not a success should possibly be rerun. Unfortunately, I could imagine weird cases with BranchOperators and tigger_rules where my proposal would be problematic. We might have to offer in time one of two cases : 1) Rerun all regardless of state. 2) Just fill the holes. But as Max said, 1) can be covered by clear + 2). So LGTM.

Comment (bolkedebruin) [reply airflow/jobs.py:1824]: Just to make sure we are clear here: we will not be exactly the same as 1.7.1. If we find out that we need this we can do a command line option (I'll create a lira for that anyway).

**Pull request closed without merging**